lib/wikidata.rb

```diff
@@ -19,10 +19,16 @@ def self.ids_from_pages(lang, titles)
         token_type: false,
       }
       response = client.action :query, page_args
-      redirected_from = Hash[(response.data['redirects'] || []).map { |h| [h['to'], h['from']] }]
-      response.data['pages'].select { |_k, v| v.key? 'pageprops' }.map do |_k, v|
-        [redirected_from[v['title']] || v['title'], v['pageprops']['wikibase_item']]
+
```

Comment:
@tmtmtmtm I've taken a stab at breaking it out into its own class (fix up: 592a14c
) but I'm not sure if I've headed in the right direction or not. I suspect the 'and' in the class name is telling me I haven't.